### DISCUSSION TITLE
Token Consumption in Anthropic's Model Context Protocol

### DISCUSSION THREAD

**Discussion by sp2935** [modelcontextprotocol/.github#64] [Q&A] [answered]
### Pre-submission Checklist

- [X] I have checked that this question would not be more appropriate as an issue in a specific repository
- [X] I have searched existing discussions and documentation for answers

### Question Category

- [ ] Protocol Specification
- [ ] SDK Usage
- [X] Server Implementation
- [ ] General Implementation
- [ ] Documentation
- [ ] Other

### Your Question

When using Anthropic's Model Context Protocol to expose resources to Claude, such as PDFs and DOCX files, does the amount of tokens in the input count each time the LLM checks the resources? If I'm using the filesystem server and Claude reads a file, that counts the first time, but is the file loaded in a way that each time it is checked, it counts as more tokens?

**Comment by jspahrsummers** [comment, marked as answer]
This is more of a Claude-specific question than an MCP question. _At the moment_, the Claude integration reads resources only when explicitly selected by the user, but this is a design decision that could change at any time.
